span.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016 Comcast Cable Communications Management, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// End Copyright
+
+package money
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+//https://github.com/Comcast/money/wiki#what-is-captured
+type span struct {
+	//the user gives us these values
+	Name    string
+	AppName string
+	TC      *TraceContext
+	Success bool
+	Code    int
+
+	//we deduce these values
+	StartTime time.Time
+	Duration  time.Duration
+	Host      string
+}
+
+//SpanReport contains parameters used for reporting back on spans
+type SpanReport struct {
+	//Name of the Span (i.e HTTPHandler)
+	Name string
+
+	//Name of the application/service running the Span (i.e. Scytale in XMiDT)
+	AppName string
+
+	//money trace context for this span
+	TC *TraceContext
+
+	//whether or not this span is defined as "successful"
+	Success bool
+
+	//Optional: status code for the operation (i.e 200 for an HTTP)
+	Code int
+}
+
+//Start begins the life of a span
+//returns a func whose calling represents the end of the span
+func Start(h http.Header) func(*SpanReport) {
```

Comment:
What's the point of passing the `http.Handler` here?  We want spans that don't correspond to full HTTP operations.

Reply:
We pass the `http.Header` because all spans generate headers that are written to it.

Other options I can think of:
1) Initialize a Span factory with such `http.Header` so we do not have to ask for it with at every span start.

2) We completely abstract away from the http package, collect all spans ourselves and leave it up to the library users to do as they want with the spans.